lib/ruby_indexer/lib/ruby_indexer/index.rb

```diff
@@ -111,6 +111,20 @@ def prefix_search(query, nesting = nil)
       results
     end
 
+    # Searches entries in the index for an exact match, intended for resolving autocomplete items. All possible matches
+    # to the prefix are returned. The return is an array of entries for the given query. For example:
+    # ## Example
+    # ```ruby
+    # # If the index has two entries for `Foo::Bar`, then:
+    # index.find("Foo::Bar")
+    # # Will return:
+    # [#<Entry::Class name="Foo::Bar">, #<Entry::Class name="Foo::Bar">]
+    # ```
+    sig { params(query: String).returns(T::Array[Entry]) }
+    def find(query)
```

Comment:
We already have the `[]` method for this, which also deletes the possible `::` prefix. Let's use that one instead.